src/ListBoxOption/index.jsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+
+import styles from './ListBoxOption.scss';
+
+export default class ListBoxOption extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.onMouseDown = this.onMouseDown.bind(this);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (!this.props.isSelected && nextProps.isSelected) {
+      this.props.onSelect();
+    }
+  }
+
+  onMouseDown() {
+    this.props.onSelect();
+  }
+
+  render() {
+    const {
+      children,
+      className,
+      index,
+      isSelected,
+      tag,
+      ...other
+    } = this.props;
+
+    return React.createElement(
+      this.props.tag,
+      {
+        'aria-selected': isSelected,
+        className: classNames(styles['list-group-item'], styles['list-group-item-action'], { active: this.props.isSelected }, className),
+        id: `list-box-option-${index}`,
```

Comment:
Maybe if `index` is not passed and it defaults to `undefined`, we should just not add an `id` attribute to the element. Otherwise, there might be a whole bunch of option elements with the same "`list-box-option-undefined`" ids.

Reply:
Agree with Tyler here.